feat: Add learning path translation (#14)
blitz-js/fr.blitzjs.com: app/pages/docs/learning-path.mdx

---
title: Blitz pas à pas 📚
sidebar_label: Blitz pas à pas
---

> Vous souhaitez perfectionner vos compétences de développement avec Blitz
> mais vous ne savez pas par où commencer ? Vous êtes arrivés sur la bonne
> page. Les liens évoqués sur cette page sont en anglais.

Reviewer: ```suggestion
Vous souhaitez perfectionner vos compétences de développement avec Blitz
mais vous ne savez pas par où commencer ? Vous êtes au bon endroit. _Les liens évoqués sur cette page sont en anglais._
```

> Inspiré par la [Frontend Developer Roadmap](https://roadmap.sh/frontend),
> ce document aborde de multiples thèmes qui touchent aux concepts & idées,
> outils & bibliothèques et histoire & acnecdotes afin de vous améliorer au
> développemnt applicatif. Cette page a été écrite pour l'usage de Blitz
> mais la plus part de ces enseignements s'appliquent universellement.

Reviewer: ```suggestion
Inspiré par la [Frontend Developer Roadmap](https://roadmap.sh/frontend),
ce document aborde de multiples thèmes qui touchent aux concepts & idées,
outils & bibliothèques et histoire & anecdotes qui vous seront utiles afin d'améliorer
vos compétences. Cette page a été écrite pour l'usage de Blitz mais la plus part de
ces enseignements s'appliquent universellement.
```

<Card type="info">

> Les liens marqué par 🥷 sont destinés aux développeurs chevronés. Si vous
> démarrez dans le monde du développement web, vous pouvez très bien les
> ignorer.

Reviewer: ```suggestion
Les liens annotés d'un 🥷 sont destinés aux développeurs chevronnés. Si vous
démarrez dans le monde du développement web, vous pouvez très bien les
ignorer.
```

</Card>

## React {#react}

> - [React for Beginners – A React.js Handbook for Front End Developers](https://www.freecodecamp.org/news/react-beginner-handbook).
>   Une grande introduction à React, y compris les composants fonctionels
>   ainsi que les Hooks de React.

Reviewer: ```suggestion
- [React for Beginners – A React.js Handbook for Front End Developers](https://www.freecodecamp.org/news/react-beginner-handbook).
  Une grande introduction à React, y compris les composants fonctionnels
  ainsi que les hooks.
```

- 📺
  [freeCodeCamp's Full React Course 2020](https://www.youtube.com/watch?v=4UZrsTqkcW4).
  Une vidéo de 10 heures d'introduction à React.
- 📺
  [The Beginner's Guide to React](https://egghead.io/courses/the-beginner-s-guide-to-react).

>   Une Courte introduction à React par Kent C. Dodds, comprenant aussi les
>   composants fonctionels.

Reviewer: ```suggestion
  Une courte introduction à React par Kent C. Dodds, comprenant aussi les
  composants fonctionnels.
```

> ## Base de données & Model de données {#databases}

Reviewer: ```suggestion
## Bases de données & Modélisation de données {#databases}
```

- [Prisma's Data Guide: Data Modeling](https://www.prisma.io/dataguide/datamodeling).
  Une introduction à la conception d'un bon schéma de base de données pour
  votre application.
- [What is Prisma?](https://www.prisma.io/docs/concepts/overview/what-is-prisma).
  Un panorama global de Prisma et ses fonctionalités.
- 🥷
  [Zero-Downtime Deployments with Data Migrations](https://engineering.wework.com/zero-downtime-deployments-with-data-migrations-2e337df48b8f).
  Comment changer votre schéma de base de données sans arrêt de service.
- Vous pouvez contribuer à cette partie en ajoutant vos moyens

>   d'apprentissage favorits via `Pull Request` ! Voici quelques idées de

Reviewer: ```suggestion
  d'apprentissage favoris via `Pull Request` ! Voici quelques idées de
```

  contenu qu'il nous manque encore :
  - un guide à propos de la création d'applications `multi-tenant`

## Intégrations {#integrations}

- Vous pouvez contribuer à cette partie en ajoutant vos moyens

>   d'apprentissage favorits via `Pull Request` ! Voici quelques idées de

Reviewer: ```suggestion
  d'apprentissage favoris via `Pull Request` ! Voici quelques idées de
```

  contenu qu'il nous manque encore :
  - comment envoyer des emails
  - comment intégrer Stripe
  - comment intégrer Paddle
  - quel fournisseur de paiement choisir

## Tâches en arrière plan & tâches Cron {#misc}

- Vous pouvez contribuer à cette partie en ajoutant vos moyens

>   d'apprentissage favorits via `Pull Request` ! Voici quelques idées de

Reviewer: ```suggestion
  d'apprentissage favoris via `Pull Request` ! Voici quelques idées de
```

  contenu qu'il nous manque encore :
  - comment les configurations Cron marchent
  - cas d'usages des tâches en arrière plan
  - comment implémenter une tâche Cron

## Tester {#testing}

> Les tests automatiques sont une bonne façon de vérifier que votre
> application marche ... et qui ne voudrait pas s'assurer que son
> application marche ?

Reviewer: ```suggestion
Les tests automatiques sont une bonne façon de vérifier que votre
application fonctionne ... et qui ne voudrait pas d'une application
fonctionnelle ?
```

- 💸 [Testing Javascript](https://testingjavascript.com). Un guide
  détaillé sur tout ce qui est relatif aux tests par Kent C. Dodds.
- Vous pouvez contribuer à cette partie en ajoutant vos moyens

>   d'apprentissage favorits via `Pull Request` ! Voici quelques idées de

Reviewer: ```suggestion
  d'apprentissage favoris via `Pull Request` ! Voici quelques idées de
```

  contenu qu'il nous manque encore :
  - une introduction à la pyramide de testing
  - comment implémenter des tests unitaires avec Jest
  - comment écrire du code facilement testable
  - un guide sur Cypress

## Architecture Logicielle {#architecture}

Au fur et à mesure que votre application grandit et se complexifie, une
structure peut devenir nécessaire. Il y a bien entendu un million de façon
de structurer votre project, voici quelques guides pour bien démarrer:

- Vous pouvez contribuer à cette partie en ajoutant vos moyens

>   d'apprentissage favorits via `Pull Request` ! Voici quelques idées de

Reviewer: ```suggestion
  d'apprentissage favoris via `Pull Request` ! Voici quelques idées de
```

  contenu qu'il nous manque encore :

>   - 🥷 TODO: un bon guilde sur l'onion et l'architecture claire

Reviewer: ```suggestion
  - 🥷 TODO: un bon guide sur l'architecture onion / hexagonale / claire
```

## Compétences de communication {#soft-skills}

> La création applicative est souvent un effort de group et requiert de la
> communication en équipe.

Reviewer: ```suggestion
La création applicative est souvent un effort de groupe et nécessite beaucoup
de travail d'équipe.
```

> - [How to Make Your Code Reviewer Fall in Love with You](https://mtlynch.io/code-review-love/).
>   Les relectures de code sont un bon outil quand elles sont bien faites
>   mais vraiment horribles lorsqu'elles ne le sont pas. Lisez cet article
>   pour améliorer grandement vos PRs !
> - Vous pouvez contribuer à cette partie en ajoutant vos moyens
>   d'apprentissage favorits via `Pull Request` !

Reviewer: ```suggestion
- [How to Make Your Code Reviewer Fall in Love with You](https://mtlynch.io/code-review-love/).
  Les revues de code sont un bon outil quand elles sont bien faites
  mais vraiment horribles lorsqu'elles ne le sont pas. Lisez cet article
  pour améliorer grandement vos PRs !
- Vous pouvez contribuer à cette partie en ajoutant vos moyens
  d'apprentissage favoris via `Pull Request` !
```

## Conception de produits {#building-products}

La conception d'un produit est bien plus que la simple conception d'une
application. Trouver le bon modèle de monétisation, évaluer les idées,
recevoir des critiques constructives, construire un panel d'utilisateurs,
tout ceci est très différent de la conception applicative. Si vous
conceptualisez un produit (une bonne partie des utilisateurs Blitz sont
entrepreneurs ! ), cette section est pour vous:

<!--alex disable dad-mom -->

- [The Mom Test](http://momtestbook.com). Un livre sur l'honnêteté lors de
  critiques d'utilisateurs.
- Vous pouvez contribuer à cette partie en ajoutant vos moyens

>   d'apprentissage favorits via `Pull Request` !

Reviewer: ```suggestion
  d'apprentissage favoris via `Pull Request` !
```
